clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java

```diff
@@ -19,15 +17,31 @@
 
     @Test
     public void checkHubConnectionState() throws Exception {
-        Transport mockTransport = new MockTransport();
+        MockTransport mockTransport = new MockTransport();
         HubConnection hubConnection = new HubConnection("http://example.com", mockTransport, true);
         hubConnection.start();
+        assertEquals(HubConnectionState.DISCONNECTED, hubConnection.getConnectionState());
+
+        mockTransport.receiveMessage("{}" + RECORD_SEPARATOR);
```

Comment:
Add "autoHandshake" to MockTransport similar to https://github.com/aspnet/SignalR/blob/release/2.2/clients/ts/signalr/tests/HubConnection.test.ts#L1103